lib/util/ip.js

```diff
@@ -0,0 +1,89 @@
+var BufferWriter = require('../encoding/bufferwriter');
+var BufferReader = require('../encoding/bufferreader');
+var constants = require('../constants');
+var serviceRegex = /^(([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])\.){3}([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5]):[0-9]+$/;
+var ipV6prefix = new Buffer(12);
+var emptyAddress = new Buffer(18);
+var EMPTY_IPV6_ADDRESS = constants.EMPTY_IPV6_ADDRESS;
+var EMPTY_IPV4_ADDRESS = constants.EMPTY_IPV4_ADDRESS;
+
+/**
+ * Maps ipv4:port to ipv6 buffer and port
+ * Note: this is made mostly for the deterministic masternode list, which are ipv4 addresses encoded as ipv6 addresses
+ * @param {string} string
+ * @return {Buffer}
+ */
+function ipAndPortToBuffer(string) {
+  if (isZeroAddress(string)) {
+    return emptyAddress.slice();
+  }
+  var split = string.split(':');
+  var addressBytes = split[0].split('.');
+  var port = Number(split[1]);
+  var addressWriter = new BufferWriter();
+  // As we are going to serialize ipv4 to ipv6 format, we need 12 zero bytes prefix
+
+  addressBytes.forEach(function(byte) {
+    var str = Number(byte).toString(16);
+    if (str.length === 1) {
+      str = '0' + str;
+    }
+    addressWriter.write(Buffer.from(str, 'hex'));
```

Comment:
Seems like it might be possible to skip the hex string conversion / strlen checks and just:

```suggestion
    addressWriter.write(Buffer.from([Number(byte)]));
```

Reply:
Nice suggestion! Actually, it can be simplified even more: 
```
var addressBuffer = Buffer.from(split[0].split('.').map(function (byte) { return Number(byte) })); 
```

Reply:
Applied. Take a look at the latest version :)